File: MAI/app/src/main/java/com/raspisanie/mai/Adapters/NewsAdapter.java
package com.raspisanie.mai.Adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.raspisanie.mai.Classes.DataModels.NewsObject;
import com.raspisanie.mai.R;

import java.util.ArrayList;

public class NewsAdapter extends RecyclerView.Adapter<NewsAdapter.NewsItem> {
    private ArrayList<NewsObject> objects;

    /**
     * Класс ViewHolder для хранения ссылок на View компоненты.
     */
    protected class NewsItem extends RecyclerView.ViewHolder {

        private TextView title;
        private TextView text;
        private TextView date;

        public NewsItem(@NonNull View itemView) {
            super(itemView);
            this.title = itemView.findViewById(R.id.title);
            this.text = itemView.findViewById(R.id.text);
            this.date = itemView.findViewById(R.id.date);
        }

        /**
         * Передача параметров в view элементы.
         * @param newsObject новость.
         */
        public void bind(NewsObject newsObject) {
            title.setText(newsObject.getTitle());
            text.setText(newsObject.getText());
            date.setText(newsObject.getDate());
        }
    }

    /**
     * Конструктор адаптера.
     * @param news список элементов для отображения.
     */
    public NewsAdapter(ArrayList news) {
        this.objects = news;
    }

    /**
     * Создание ViewHolder списка c пустыми элементами.
     * @param viewGroup
     * @param i позиция элемениа.
     * @return элемент списка с View элементами.
     */
    @NonNull
    @Override
    public NewsAdapter.NewsItem onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_news_list, viewGroup, false);
        return new NewsAdapter.NewsItem(view);
    }

    /**
     * Передача параметров в элемент списка итемов.
     * @param item эоемент списка.
     * @param i
     */
    @Override
    public void onBindViewHolder(@NonNull NewsAdapter.NewsItem item, int i) {
        item.bind(objects.get(i));
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public int getItemCount() {
        return objects.size();
    }
}

File: MAI2/mai2/src/main/java/com/raspisanie/mai/ui/main/info/news/NewsAdapter.kt
package com.raspisanie.mai.ui.main.info.news

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.raspisanie.mai.R
import com.raspisanie.mai.extesions.fromFormatToFormat
import com.raspisanie.mai.models.local.NewsLocal
import kotlinx.android.synthetic.main.item_news.view.*


class NewsAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private val list = mutableListOf<NewsLocal>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ItemHolder(
                LayoutInflater.from(parent.context).inflate(R.layout.item_news, parent, false)
        )
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is ItemHolder) holder.bind(list[position], position)
    }

    override fun getItemCount(): Int = list.size

    fun addAll(list: MutableList<NewsLocal>) {
        this.list.clear()
        this.list.addAll(list)
        notifyDataSetChanged()
    }

    inner class ItemHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @SuppressLint("SetTextI18n")
        fun bind(news: NewsLocal, position: Int) {
            with(itemView) {
                tvTitle.text = "${news.title} "
                tvAuthor.text = news.author
                tvText.text = news.text
                tvDate.text = news.createdAt.split(".")[0].fromFormatToFormat("yyyy-MM-dd HH:mm:ss", "HH:mm dd.MM.yyyy")
                ivWarning.visibility = if (news.warning) View.VISIBLE else View.GONE
                //vDivider.visibility =
                //if ((position < list.lastIndex && list[position + 1].warning) || news.warning) View.INVISIBLE else View.VISIBLE
            }
        }
    }
}